docs/layer_util.js

```diff
@@ -610,24 +621,47 @@ function removeScoreLayer() {
   redrawLayers();
 }
 
-const mapLoadingFinished = () => loadingElementFinished(mapContainerId);
+/** Notes that a layer has finished rendering on the map. */
+function mapLoadingFinished() {
+  loadingElementFinished(mapContainerId);
+}
 
 /**
  * Notes that an element has started loading, and add a handler to the Promise
- * to note when it finishes.
+ * to note when it finishes. Also handle errors if the Promise fails.
  * @param {Promise} promise
+ * @param {Object} layerInfoForErrors Data for layer in same format as coming
+ *     from Firestore. Must have `display-name` and `index` properties. Only
+ *     used in case of errors.
  * @return {Promise} wrappedPromise
  */
-function wrapPromiseLoadingAware(promise) {
+function wrapPromiseLoadingAware(promise, layerInfoForErrors) {
   addLoadingElement(mapContainerId);
-  return promise.then(mapLoadingFinished);
+  const {index, 'display-name': displayName} = layerInfoForErrors;
+  return promise
+      .catch((err) => {
+        showError(
+            'Error with layer ' + displayName + ', ' + err,
+            'Error loading layer ' + displayName);
+        $('#' + getCheckBoxRowId(index))
+            .prop(
+                'title',
+                'Error showing layer. If you believe the layer is there, try ' +
+                    'refreshing the page')
+            .css('text-decoration', 'line-through');
+        $('#' + getCheckBoxId(index))
+            .prop('checked', false)
+            .prop('disabled', true);
```

Comment:
Woo great I'm seeing this after my comment in the other PR I think this is a lot better than just getting rid of the box entirely.

Reply:
This is different because nothing happens when you click on it, so disabling is easy.